environment/uid_windows.go

```diff
@@ -0,0 +1,25 @@
+package environment
+
+import (
+	"os/user"
+	"regexp"
+)
+
+var sidExp = regexp.MustCompile(`([\d\D]+-)(?P<uid>\d+)`)
+
+func initUid(envStorage EnvStorage) {
+	// under native windows defaults to using
+	// root inside containers for kool managed images
+	envStorage.Set("UID", uid())
+}
+
+func uid() string {
+	current, _ := user.Current()
+	match := sidExp.FindStringSubmatch(current.Uid)
+
+	results := map[string]string{}
+	for i, name := range match {
+		results[sidExp.SubexpNames()[i]] = name
+	}
+	return results["uid"]
```

Comment:
what is the change we run this in a different version of windows, and for some reason we do not get a proper "uid" match from FindStringSubmatch above?

I guess we should here check IF the "uid" key exists, and then return its value; other wise I guess we should not break but instead we can default to the old behaviour?

Reply:
I did it.
I tried to do a unit test to get this error, but I couldn't, I commented where the test failed. 
I wanted to simulate a case of divine work. 😄